plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -344,7 +344,10 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         }
 
         if (type instanceof TimeType) {
-            // TODO https://github.com/trinodb/trino/issues/6956
+            TimeType timeType = (TimeType) type;
+            int precision = min(timeType.getPrecision(), MAX_SUPPORTED_TIMESTAMP_PRECISION);
```

Comment:
This is TIME not TIMESTAMP. We should have two separate MAX_* constants. Or rename `MAX_SUPPORTED_TIMESTAMP_PRECISION` to `MAX_SUPPORTED_TEMPORAL_PRECISION`

Reply:
Renamed to `MAX_SUPPORTED_TEMPORAL_PRECISION`!